moveit_ros/hybrid_planning/local_planner/local_planner_component/src/local_planner_component.cpp

```diff
@@ -279,6 +282,15 @@ void LocalPlannerComponent::executeIteration()
         local_planning_goal_handle_->publish_feedback(local_planner_feedback_);
       }
 
+      // Publish at the period given in the previous waypoint
+      // If initialized. Else, publish immediately.
+      if (prev_waypoint_duration_)
```

Comment:
Better match `time_from_start` with the actual current time and sleep for the diff duration to rule out lagging behind.